osquery/core/sqlite_util.cpp

```diff
@@ -63,4 +63,56 @@ int query_data_callback(void* argument,
   (*qData).push_back(r);
   return 0;
 }
+
+Status getQueryColumns(const std::string& q, tables::TableColumns& columns) {
+  sqlite3* db = createDB();
+  Status status = getQueryColumns(q, columns, db);
+  sqlite3_close(db);
+  return status;
+}
+
+Status getQueryColumns(const std::string& q,
+                       tables::TableColumns& columns,
+                       sqlite3* db) {
+  int rc;
+
+  // Will automatically handle calling sqlite3_finalize on the prepared stmt
+  // (Note that sqlite3_finalize is explicitly a nop for nullptr)
+  std::unique_ptr<sqlite3_stmt, decltype(sqlite3_finalize)*>
```

Comment:
Aside: do you think some of the templating in `osquery/core/virtual_tables.h` could make use of decltypes? It would be great to reduct the number of template symbols. I was also considering something like: http://en.wikipedia.org/wiki/Curiously_recurring_template_pattern for TablePlugin inheritors. We could potentially remove the requirement for templating "every" sqlite_module method.